spec/models/pxe_server_spec.rb

```diff
@@ -310,4 +312,65 @@ def file_write(file, contents)
       expect(@pxe_server.discovered_pxe_images).to eq([@discovered_image])
     end
   end
+
+  describe '#ensure_menu_list' do
+    context 'when creating server' do
+      subject { FactoryBot.build(:pxe_server) }
+      it 'existing menus remain' do
+        subject.ensure_menu_list(%w[new])
+        expect(PxeMenu.count).to eq(1)
```

Comment:
Should these tests be looking at `subject.pxe_menus.count`?

Reply:
Thanks, added those relations checks in addition to global checks. I would keep global checks as well to ensure nothing remains in database outside the relations.